src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -33,7 +33,6 @@ internal partial class LanguageParser : SyntaxParser
         private int _recursionDepth;
         private TerminatorState _termState; // Resettable
         private bool _isInTry; // Resettable
-        private bool _checkedTopLevelStatementsFeatureAvailability; // Resettable
```

Comment:
`// resettable` was not true.  So there was likely a contex-sensitivity parsing bug lurking there with incremental and top-level statements.